app/src/utils/routingUtils.ts

```diff
@@ -1,19 +1,40 @@
 export const RETURN_URL_URL_PARAM = "returnUrl";
 
 /**
- * Gets the current url path and query string, excluding the domain and protocol.
+ * Creates a URL to redirect to the login page with the current path as the return URL
  */
-const getCurrentUrlPath = (): string => {
-  return window.location.pathname + window.location.search;
-};
+export function createLoginRedirectUrl() {
```

Comment:
I think a lot of the logic in this function is handled by the URL class constructor already

https://developer.mozilla.org/en-US/docs/Web/API/URL_API/Resolving_relative_references

Reply:
These API seem like they require a base URL. Can they also handle concatenating two paths?

Reply:
const url = new URL(path, window.Config.basename || window.location.origin)

This gives you access to a ton of properties for free, including search params, will correctly concatenate even if the path already has a baseUrl, and more. I would encourage using something like this rather than manually splitting and concatenating urls